pulp_rpm/app/migrations/0012_rpm_evr_extension.py

```diff
@@ -0,0 +1,58 @@
+
+# This Migration was _not_ automatically generated.
+# When regenerating the migrations ever, this one _must_ be preserved.
+
+from django.db import migrations
+from django.contrib.postgres.operations import CreateExtension
+from pulp_rpm.app.models import RpmVersionField
+
+
+sql = """
+update rpm_package set evr = (
+  select ROW(coalesce(epoch::numeric,0),
+             rpmver_array(version)::evr_array_item[],
+             rpmver_array(release)::evr_array_item[])::evr_t
+  );
+
+-- create evr_t on insert, so it matches the provided E/V/R cols
+CREATE TRIGGER evr_insert_trigger
+  BEFORE INSERT
+  ON rpm_package
+  FOR EACH ROW
+  EXECUTE PROCEDURE evr_trigger();
+
+-- create evr_t on update, so it continues to match the provided E/V/R cols,
+-- but only if Something Changed
+CREATE TRIGGER evr_update_trigger
+  BEFORE UPDATE OF epoch, version, release
+  ON rpm_package
+  FOR EACH ROW
+  WHEN (
+    OLD.epoch IS DISTINCT FROM NEW.epoch OR
+    OLD.version IS DISTINCT FROM NEW.version OR
+    OLD.release IS DISTINCT FROM NEW.release
+  )
+  EXECUTE PROCEDURE evr_trigger();
+"""
```

Comment:
would it be better to only have the new "evr" field in the database? I think, you can add properties of the old names to the model to keep it up to date.

Reply:
I don't think so.  Strictly speaking I don't think the update trigger should ever be necessary because there's no way to modify content units anyway.  But also, we need to provide those values separately for content/packages/ anyway, and I don't think we should change it mid-stream.  Also I'm not sure how writing to this field from Django would work.